sdk/identity/azure-identity/azure/identity/_credentials/vscode_credential.py

```diff
@@ -45,9 +47,8 @@ def get_token(self, *scopes, **kwargs):
 
         refresh_token = get_credentials()
```

Comment:
Does this get the refresh token from the protected store every time? Perhaps we should only retrieve from the store once?

Reply:
I think the question is will this change? e.g. if user logs out on VS code, what's our expected behavior?

Reply:
I think the behavior should be that once we authenticate we continue to use that account even if the VS code user changes during the execution of the process, any other behavior is too hard to reason about. That being said this is also an issue in the .NET implementation now, so if we can't fix it today, we can create an issue to ensure we fix it by GA.

Reply:
Updaed.

Reply:
To be clear, this update doesn't implement the behavior Scott described. The credential can still change refresh token and thus identity. To behave as described, it would need to read the refresh token exactly once.